### ISSUE TITLE
Implement import conventions

### ISSUE THREAD

**Issue by edgarrmondragon** [charliermarsh/ruff#1004]
There's https://github.com/joaopalmeiro/flake8-import-conventions to serve as inspiration, though I'm not sure if I'd like to replicate the behavior 100%.

It has an error code for each convention:

```python
name2asname: Dict[str, str] = {
    "altair": "alt",   # IC001
    "numpy": "np",     # IC002
    "pandas": "pd",    # IC003
    "seaborn": "sns",  # IC004
}
```

The biggest problem with this approach is that it's not possible to expand with custom conventions without forking the repo.

One alternative would be to use a single error code for all conventions, default to the ones above, and allow users to configure a custom mapping:

```toml
[tool.ruff.import_conventions.aliases]
pulumi_aws = "aws"
```

**Comment by charliermarsh** [comment]
Yeah I like what you outlined above: one rule ("Invalid alias" or whatever), with some defaults, that can be overridden.

**Comment by edgarrmondragon** [comment]
Cool, I'll work on the implementation then 😄
